### PR TITLE
add a test of available_linters(packages=)

### LINKED ISSUE #933: Test packages= argument of available_linters by adding a test package
So far all tests are just of the default `packages="lintr"`, we should add a test of a test package that has an `inst/lintr/linters.csv` file.

### PR DESCRIPTION
**Pull request by MichaelChirico**
Closes #933.

I wrote a few different approaches for constructing the difference between the csv on disk & the expected output, nothing seemed particularly clean, but feel free to chime in if you've got a preference. Fundamentally it's challenging that `write.csv()`/`read.csv()` don't support list columns, hence a need to massage a bit.

### CODE REVIEW

File: tests/testthat/test-linter_tags.R [hunk 1]
```diff
@@ -233,3 +233,35 @@ test_that("available_linters gives precedence to included tags", {
...
+    system.file = function(..., package) file.path(db_loc, package, ...)
+  )
+
+  custom_db <- data.frame(
+    linter = c("linter1", "linter2"),
+    package = "myPkg",
+    tags = c("tag1 tag2", "tag1 tag3")
+  )
+  custom_db$tags <- strsplit(custom_db$tags, " ", fixed = TRUE)
+
```
Comment (MichaelChirico) [inline tests/testthat/test-linter_tags.R:249]:
```suggestion

```

**Review by AshesITR — APPROVED** [review]

File: tests/testthat/test-linter_tags.R [hunk 2]
```diff
@@ -233,3 +233,39 @@ test_that("available_linters gives precedence to included tags", {
...
+    file.path(custom_loc, "linters.csv")
+  )
+
+  expect_identical(
+    available_linters(packages = "myPkg"),
+    custom_db
+  )
+
+  # edge case
+  write_csv(custom_db[0, c("linter", "tags")], file.path(custom_loc, "linters.csv"))
```
Comment (MichaelChirico) [inline tests/testthat/test-linter_tags.R:266]:
```suggestion
  write_csv(custom_db[0L, c("linter", "tags")], file.path(custom_loc, "linters.csv"))
```

File: tests/testthat/test-linter_tags.R [hunk 3]
```diff
@@ -233,3 +233,39 @@ test_that("available_linters gives precedence to included tags", {
...
+  expect_identical(
+    available_linters(packages = "myPkg"),
+    custom_db[0, ]
```
Comment (MichaelChirico) [inline tests/testthat/test-linter_tags.R:269]:
```suggestion
    custom_db[0L, ]
```

**Review by AshesITR — APPROVED** [review]
